liblwgeom/lwcollection.c

```diff
@@ -214,24 +214,32 @@ LWCOLLECTION* lwcollection_add_lwgeom(LWCOLLECTION *col, const LWGEOM *geom)
 	/* See http://trac.osgeo.org/postgis/ticket/2933 */
 	/* Make sure we don't already have a reference to this geom */
 	{
-	int i = 0;
-	for ( i = 0; i < col->ngeoms; i++ )
-	{
-		if ( col->geoms[i] == geom )
+		uint32_t i = 0;
+		for ( i = 0; i < col->ngeoms; i++ )
 		{
-            lwerror("%s [%d] found duplicate geometry in collection %p == %p", __FILE__, __LINE__, col->geoms[i], geom);
-			LWDEBUGF(4, "Found duplicate geometry in collection %p == %p", col->geoms[i], geom);
-			return col;
+			if ( col->geoms[i] == geom )
+			{
+				lwerror("%s [%d] found duplicate geometry in collection %p == %p", __FILE__, __LINE__, col->geoms[i], geom);
+				return col;
+			}
 		}
 	}
-	}
 #endif
 
 	col->geoms[col->ngeoms] = (LWGEOM*)geom;
 	col->ngeoms++;
 	return col;
 }
 
+LWCOLLECTION*
+lwcollection_concat_in_place(LWCOLLECTION* col1, const LWCOLLECTION* col2)
```

Comment:
Add source comment to document what this function does, and what responsibilities of caller are. (It looks like caller would need to release col2 after calling?)